core/src/messages/scheduler.rs

```diff
@@ -36,9 +36,9 @@ pub struct ScheduleRequest {
 }
 
 impl ScheduleRequest {
-    pub fn schedule(&self, drone_id: DroneId) -> SpawnRequest {
+    pub fn schedule(&self, drone_id: &DroneId) -> SpawnRequest {
```

Comment:
So, this is a rust idiom right? to generally borrow in function params as much as possible? why?

Reply:
Yes. Before, `DroneId` wrapped a `u32`, which is cheap to copy, so `DroneId` implemented `Copy`, and the compiler would make copies of it freely. That way, you can pass it to functions like this without the caller losing ownership of the value.

With this update, `DroneId` wraps a `String`, so it does not implement `Copy`. Our choice here would either be to keep this function signature as-is and call `.clone()` on every `DroneId` we pass in (except where we really do want to transfer ownership), or do what I do here and take a reference.

Generally, my decision between the two is:
- If an argument implements `Copy`, the signature should receive it by value rather than as a reference.
- If an argument does not implement `Copy` and the function does not make a copy of it, take the value by reference.
- If an argument does not implement `Copy` and the function needs to make a copy of it (e.g. to put it into a data structure and return it), it's a judgement call; usually I receive it by value and allow the caller to `clone()` if needed, but if I know that the caller will always need to `clone()`, I might take a reference and bake the cloning behavior into the function.

Reply:
ahhh that makes a lot of sense :+1: